snapshot/snapshot.go

```diff
@@ -146,8 +161,10 @@ func NewSnapshotter(ctx context.Context, root string, targetFs FileSystem, opts
 		userxattr:   userxattr,
 	}
 
-	if err := o.restoreRemoteSnapshot(ctx); err != nil {
-		return nil, errors.Wrap(err, "failed to restore remote snapshot")
+	if !config.enableFuseManager {
+		if err := o.restoreRemoteSnapshot(ctx); err != nil {
+			return nil, errors.Wrap(err, "failed to restore remote snapshot")
+		}
```

Comment:
Can we keep snapshotter agnostic about fuse manager?
Thus this option should be something like `enableRestoreSnapshots`.

Reply:
Replaced with `enableRestoreSnapshots`